### PR TITLE
System check for KIWI_TENANTS_DOMAIN ENV

### PR DESCRIPTION
**Pull request by schwarzkrieger**
refs #140

### CODE REVIEW

File: tcms_tenants/checks/tenants_env.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+@checks.register
+def tenants_env_check(app_configs, **kwargs):
+    errors = []
+    if os.environ.get('KIWI_TENANTS_DOMAIN') is None:
+        errors.append(
+            checks.Error(
+                level='CRITICAL',
+                msg='KIWI_TENANTS_DOMAIN environment variable is not set!',
+                hint='You need this variable set to run tenancy enabled system.',
```
Comment (atodorov) [inline tcms_tenants/checks/tenants_env.py:14]: Change to something like `This variable is mandatory, see https://github.com/kiwitcms/tenants#installation`

File: tcms_tenants/checks/tenants_env.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import os
+
+from django.core import checks
+
+
+@checks.register
+def tenants_env_check(app_configs, **kwargs):
+    errors = []
+    if os.environ.get('KIWI_TENANTS_DOMAIN') is None:
```
Comment (atodorov) [inline tcms_tenants/checks/tenants_env.py:9]: `if not os.environ.get` to catch the case where this is an empty string.

File: tcms_tenants/checks/tenants_env.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import os
```
Comment (atodorov) [inline tcms_tenants/checks/tenants_env.py:1]: Add the copyright header with your name as author, see `admin.py` for example.

**Review by atodorov — CHANGES_REQUESTED** [review]
- Minor changes requested
- Let's use `Close ` for the original issue. I don't remember what I meant with `1 tenant/domain`. 
- This new check will also need an automated test. Probably best to add as an additional check inside github actions.

File: tcms_tenants/checks/tenants_env.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+from django.core import checks
+
+
+@checks.register
```
Comment (atodorov) [inline tcms_tenants/checks/tenants_env.py:6]: This isn't going to register the check automatically, see `apps.py` in https://github.com/kiwitcms/Kiwi/pull/2663

**Review by atodorov — CHANGES_REQUESTED** [review]

File: tcms_tenants/checks.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# Copyright (c) 2019-2020 Ivajlo Karabojkov <karabojkov@kit.bg>
```
Comment (atodorov) [inline tcms_tenants/checks.py:1]: That's 2021-2022, isn't it.

File: tcms_tenants/checks.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+# Licensed under the GPL 3.0: https://www.gnu.org/licenses/gpl-3.0.txt
+
+
+import os
+
+from django.core import checks
+
+
+@checks.register
```
Comment (atodorov) [inline tcms_tenants/checks.py:11]: this decorator doesn't do anything.

**Review by atodorov — CHANGES_REQUESTED** [review]
Some changes requested.

Still needs tests, see https://github.com/kiwitcms/Kiwi/pull/2663

FTR b/c this check function doesn't check any internals you can use the following approach: 
- using standard Django TestCase classes to exercise the function directly and assert on the expected result. Use `patch.dict` to manipulate the environment, see https://github.com/kiwitcms/tcms-api/blob/master/tests/test_version.py#L17
- one test to make sure this check is actually running, aka the registration is working.

File: tcms_tenants/apps.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# pylint: disable=import-outside-toplevel
+from django.apps import AppConfig as DjangoAppConfig
+from django.core.checks import register
+
+
+class AppConfig(DjangoAppConfig):
+    name = "tcms_tenants"
```
Comment (atodorov) [inline tcms_tenants/apps.py:7]: Hmm. I can't find how this app currently gets into INSTALLED_APPS. I don't think it is listed there and it is surprising that it even loads and works. Need to investigate more.

Comment (atodorov) [reply tcms_tenants/apps.py:7]: Found it https://github.com/kiwitcms/Kiwi/blob/master/tcms/settings/common.py#L293

Comment (atodorov) [inline tcms_tenants/apps.py:7]: This attribute shouldn't be necessary. Try removing it and we'll test without it.

Comment (schwarzkrieger) [reply tcms_tenants/apps.py:7]: An exception occurs without it:
django.core.exceptions.ImproperlyConfigured: 'tcms_tenants' must supply a name attribute.

File: Makefile [hunk 8]
```diff
@@ -14,6 +14,7 @@ test:
 	fi
 	
 	PYTHONPATH=.:$(KIWI_INCLUDE_PATH) EXECUTOR=standard PYTHONWARNINGS=d AUTO_CREATE_SCHEMA='' \
+        KIWI_TENANTS_DOMAIN='test.com' \
 	    coverage run --include "tcms_tenants/*.py" \
```
Comment (atodorov) [inline Makefile:18]: This change is needed b/c of the fact that checks are executed automatically upon migrate (which is triggered by test), correct ? 

We'll also need an additional test, just to make sure the registration portion is working. I think the easiest way is probably to duplicate this line and call `manage.pt check`, not set the ENV variable and grep for the expected error message.

**Review by atodorov — CHANGES_REQUESTED** [review]
Overall looks good to me, needs a bit more work on the test side.

**Pull request merged**